core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -340,14 +440,119 @@ public boolean dropNamespace(Namespace namespace) throws NamespaceNotEmptyExcept
   @Override
   public boolean setProperties(Namespace namespace, Map<String, String> properties) throws
       NoSuchNamespaceException {
-    throw new UnsupportedOperationException(
-        "Cannot set properties " + namespace + " : setProperties is not supported");
+    if (!namespaceExists(namespace)) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+
+    if (properties == null || properties.isEmpty()) {
+      throw new IllegalArgumentException("Cannot setProperties to a namespace with null or empty properties");
+    }
```

Comment:
Nit: This could probably be `Preconditions.checkArgument` or some variant of that.